packages/ethereum-storage/src/ethereum-entries-to-ipfs-content.ts

```diff
@@ -0,0 +1,251 @@
+
+  // Try to read the hashes on IPFS
+  // The operation is done at least once and retried depending on the readOnIPFSRetry config
+  for (let tryIndex = 0; tryIndex < 1 + getMaxIpfsReadRetry(); tryIndex++) {
+    // Reset for each retry
+    ipfsConnectionErrorCount = 0;
+
+    if (tryIndex > 0) {
+      logger.debug(`Retrying to read hashes on IPFS`, ['ipfs']);
+    }
+
+    allIpfsContentOrErrors = await Bluebird.map(
+      ethereumEntriesToProcess,
+      // Read hash on IPFS and retrieve content corresponding to the hash
+      // Reject on error when no file is found on IPFS
+      // or when the declared size doesn't correspond to the size of the content stored on ipfs
+      async (ethereumEntry: StorageTypes.IEthereumEntry) => {
+        return getIpfsContent(ethereumEntry, tryIndex + 1, ipfsManager, logger);
+      },
+      {
+        concurrency: maxConcurrency,
+      },
+    );
+
+    // flush the list of entries to process
+    ethereumEntriesToProcess = [];
+
+    // Store found hashes in entries
+    // The hashes to retry to read are the hashes where readHashOnIPFS returned null
+    for (const { ipfsContent, entryWithError } of allIpfsContentOrErrors) {
+      if (ipfsContent) {
+        // content found and not error
+        finalIpfsContents.push(ipfsContent);
+      } else if (entryWithError) {
+        const errorType = entryWithError.error!.type;
+        if (errorType === StorageTypes.ErrorEntries.INCORRECT_FILE) {
+          incorrectFileCount++;
+          // no retry needed, just store it
+          await ignoredDataIdsIndex.save(
+            entryWithError.hash,
+            entryWithError.error!.message,
+            false,
+          );
+        } else if (errorType === StorageTypes.ErrorEntries.WRONG_FEES) {
+          wrongFeesCount++;
+          // no retry needed, just store it
+          await ignoredDataIdsIndex.save(
+            entryWithError.hash,
+            entryWithError.error!.message,
+            false,
+          );
+        } else if (errorType === StorageTypes.ErrorEntries.IPFS_CONNECTION_ERROR) {
+          ipfsConnectionErrorCount++;
+          // push it for a retry
+          ethereumEntriesToProcess.push(entryWithError);
+        } else {
+          throw new Error(`Unexpected Error for the hash: ${entryWithError.hash}, ${entryWithError.error?.type}, ${entryWithError.error?.message}`);
+        }
+      }
+    }
+
+    successCount = finalIpfsContents.length;
+
+    logger.debug(`${successCount}/${totalCount} retrieved dataIds after try ${tryIndex + 1}`, ['ipfs']);
+
+    if (tryIndex === 0) {
+      successCountOnFirstTry = successCount;
+    }
+  }
+
+  // Save the entries not successfully retrieved after the retries
+  for (const entryRemaining of ethereumEntriesToProcess) {
+    // store the ipfs ignored after the retried
+    await ignoredDataIdsIndex.save(
+      entryRemaining.hash,
+      entryRemaining.error!.message,
+      true,
```

Comment:
I'm not a big fan of booleans as arguments because of this. By reading this file I have no idea what true means. 
I would suggest having a thin function over save, like `saveWithRetry`, that will set this argument to true (and maybe `saveWithoutRetry` for the other cases, but I guess in this context we could just use the default value).

Reply:
Since TS and an IDE help a bit with that, I'm not going to say it's necessary to merge the PR. But it's a nice to have.

Reply:
Spoiler alert: This will be removed in the next PR.
But I keep this in mind!